site/content/integrate/apps/deployment/_index.md

```diff
@@ -52,40 +49,128 @@ For a go app the manifest snippet would look like this:
         }
     ]
 }
-``` 
+```
+
+### Setting up your Mattermost instance and AWS for deploying custom apps
+
+Note that custom apps are not supported in the Mattermost Cloud - they're only used for developer testing convenience. Provisioning in the AWS cloud environment is done by the **appsctl** tool after completing the environment set up.
+
+We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3 (not implemented yet), or from a URL (not implemented yet). Provisioning consists of three parts which will be completed via the **appsctl** tool:
+
+1. Creating the lambda functions with appropriate policies.
+2. Storing static assets in the dedicated S3 bucket.
+3. Storing the app’s manifest file in the same dedicated S3 bucket.
+
+#### Mattermost setup
+
+Follow the Prerequisites section of the [JS](../quick-start-js) or [GO](../quick-start-go) Quick Start Guides to set up your Mattermost instance. Ensure that you install the [apps plugin](https://developers.mattermost.com/integrate/apps/quick-start-js/#install-the-apps-plugin) as part of this process.
+
+#### AWS setup
+
+##### 1. Create AWS S3 bucket
+
+You will need to create an S3 bucket within AWS or use an existing bucket.
+
+1. Visit https://s3.console.aws.amazon.com
+1. Create a bucket.
+1. **Bucket name**: Give your bucket a name.
```

Comment:
Should we provide suggestions?

Reply:
* **Pros** 
  * consitent naming convention

* **Cons**
  * naming does not matter
  * a user can use an existing bucket
  * some might assume the name itself has significance based on the example

Dylan and I discussed this for a decent while and voted against :)

Reply:
Makes sense :+1: